lib/identity_cache/query_api.rb

```diff
@@ -24,13 +24,19 @@ def fetch_by_id(id, includes: nil)
         return unless id
         record = if should_use_cache?
           require_if_necessary do
-            object = nil
-            coder = IdentityCache.fetch(rails_cache_key(id)){ instrumented_coder_from_record(object = resolve_cache_miss(id)) }
-            object ||= instrumented_record_from_coder(coder)
-            if object && object.id != id
-              IdentityCache.logger.error "[IDC id mismatch] fetch_by_id_requested=#{id} fetch_by_id_got=#{object.id} for #{object.inspect[(0..100)]}"
+            cache_key = rails_cache_key(id)
+            begin
```

Comment:
I find both `fetch_by_id` and the multi version are getting too chunky, is there a way to factor out the new logic into a function that we can call here ?